nemo_curator/utils/distributed_utils.py

```diff
@@ -267,6 +269,23 @@ def _set_torch_to_use_rmm():
     torch.cuda.memory.change_current_allocator(rmm_torch_allocator)
 
 
+def select_columns(
+    df: Union[dd.DataFrame, dask_cudf.DataFrame],
+    columns: List[str],
+    filetype: Literal["jsonl", "json", "parquet"],
+    add_filename: bool,
+) -> Union[dd.DataFrame, dask_cudf.DataFrame]:
+    # We exclude parquet because the parquet readers already support column selection
+    if filetype in ["jsonl", "json"] and columns is not None:
```

Comment:
Your comment in the README which says:

"`df.columns` does not show `filename` when `columns` are provided, `filetype="parquet"`, and `add_filename=True`"

is because of this?

Reply:
No, it was a very interesting issue now that we figured thanks to @rjzamora 

> You are basically running into a case where from_map is trying (but failing) to be "smart". When the callable argument to from_map has a columns= argument in its signature, we end up with a [FromMapProjectable](https://github.com/dask/dask-expr/blob/77d0f895238b4c90063e40b75ffb0e45564afd7c/dask_expr/io/io.py#L297)  expression. This is because dask-expr assumes that having this argument means that your operation supports column projection (and that passing columns=["text", "id"] means that the output of that operation must contain those exact columns.
If you don't want this behavior, you should call the argument something else. e.g.:
def read_single_partition_pq(files: List[str], io_columns: Optional[List[str]] = None):